apollo-federation/src/sources/connect/json_selection/README.md

```diff
@@ -82,9 +82,10 @@ below.
 
 ```ebnf
 JSONSelection        ::= PathSelection | NamedSelection*
-SubSelection         ::= "{" NamedSelection* "}"
+SubSelection         ::= "{" OutputTypeAnnotation? NamedSelection* "}"
+OutputTypeAnnotation ::= "<" Identifier ">"
 NamedSelection       ::= NamedPathSelection | PathWithSubSelection | NamedFieldSelection | NamedGroupSelection
-NamedPathSelection   ::= Alias PathSelection
+NamedPathSelection   ::= (Alias | "...") PathSelection
 NamedFieldSelection  ::= Alias? Key SubSelection?
 NamedGroupSelection  ::= Alias SubSelection
```

Comment:
Here's the only new JSONSelection syntax we need: `{ <Type> ... }` annotations and `...` before inlined `NamedPathSelection` items.